### DISCUSSION TITLE
Need Help on add team as reviewers with GitHub action

### DISCUSSION THREAD

**Discussion by salini8642** [community/community#63927] [Actions] [answered]
I have a team with write permissions and created workflow file it will executed successfully but here the issue is I'm unable to see the my team name in the reviewer session and also they didn't receiving the notifications. Can you please guide me.
Workflow file:
on:
  pull_request:

 

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - name: Add Reviewers
      uses: uhg-actions/add-reviewers@v1
      with:
        token: "${{ secrets.GITHUB_TOKEN }}" # "${{ secrets.ACCESS_TOKEN }}"
        reviewers: ghec_deepa_admin
        re-request-when-approved: true
        re-request-when-changes-requested: true

**Comment by nirav-gajera** [comment]
The issue is that you are using the `reviewers` keyword to specify the team name, but this keyword only accepts individual usernames. To specify a team, you need to use the `reviewer-teams` keyword.

Here is the updated workflow :

```
on:
pull_request:

jobs:
build:
runs-on: ubuntu-latest
steps:
- name: Add Reviewers
uses: uhg-actions/add-reviewers@v1
with:
token: "${{ secrets.GITHUB_TOKEN }}" # "${{ secrets.ACCESS_TOKEN }}"
reviewer-teams: ghec_deepa_admin
re-request-when-approved: true
re-request-when-changes-requested: true
```

**Reply by salini8642** [reply]
when I added this way reviewer-teams: ghec_deepa_admin it giving like invalid input

**Comment by nirav-gajera** [comment]
try with quotes:  `reviewer-teams: "ghec_deepa_admin"`

**Reply by salini8642** [reply]
tried that way also or can you please suggest with new configuration for add team as reviewers with github actions instead of using github API

**Reply by salini8642** [reply]
what you suggested troubleshoot steps I have already configured everything but I don't know here what is the issue and I have used chatGPT also but issue is not find out yet. If anything I missed

**Reply by salini8642** [reply]
I need one confirmation my org is maintaining some actions related repos but I'm trying to add team as reviewer action it is not available in my org actions now what I'm trying to achieving is possible without team reviewer action

**Reply by nirav-gajera** [reply, marked as answer]
try  GitHub Actions instead of using the GitHub API.

ex:

```
on:
pull_request:
  types: [opened, reopened]

jobs:
  add-reviewers:
    runs-on: ubuntu-latest
    steps:
      - name: Add Reviewers
        uses: actions/github-script@v2
        with:
          script: |
            const team = 'ghec_deepa_admin';
            const token = process.env.GITHUB_TOKEN;

            const response = await github.rest.pulls.addReviewers({
              pull_id: context.payload.pull_id,
              team_reviewers: [team],
              token
            });

            console.log(response);
```

This workflow uses the `actions/github-script@v2` action to run a GitHub Script. The GitHub Script uses the GitHub API to add the team `ghec_deepa_admin` as reviewers to the pull request.

**Reply by salini8642** [reply]
this reading node modules but in my repo I'm not using node modules

**Reply by nirav-gajera** [reply]
```

on:
pull_request:
  types: [opened, reopened]

jobs:
  add-reviewers:
    runs-on: ubuntu-latest
    steps:
      - name: Add Reviewers
        uses: stefanzweifel/add-reviewers@v1
        with:
          team: 'ghec_deepa_admin'
          token: ${{ secrets.GITHUB_TOKEN }}
```
This workflow uses the stefanzweifel/add-reviewers@v1 action, which is a simpler action that does not require node modules.

**Reply by salini8642** [reply]
My issue is resolved and Thank you for your suggestion @nirav-gajera

**Comment by oftharsh** [comment]
Please mark this as answer.

It looks like you're using the `uhg-actions/add-reviewers` action to automatically add reviewers to pull requests when a workflow is triggered. However, you mentioned that the team name is not appearing in the reviewer session and team members are not receiving notifications. Here are a few steps to troubleshoot and address the issue:

1. **Check Team Name**: Ensure that the `ghec_deepa_admin` value is the correct name of the team you want to add as reviewers. Double-check the spelling and capitalization of the team name.

2. **Team Permissions**: Make sure that the team `ghec_deepa_admin` has the appropriate permissions to be added as reviewers. They should have at least "Read" permission on the repository for the action to work.

3. **Secrets Configuration**: Verify that the `GITHUB_TOKEN` secret is correctly set in your repository settings. The `secrets.GITHUB_TOKEN` token is automatically provided by GitHub to workflows and should have the necessary permissions to perform actions like adding reviewers.

4. **Workflow Trigger**: Ensure that the workflow is being triggered correctly when a pull request is opened. You can check the Actions tab in your repository to see if the workflow is being executed and if there are any errors or issues.

5. **Action Output**: Check the output of the workflow run to see if there are any error messages or warnings related to the `uhg-actions/add-reviewers` action. This might provide more information about why the reviewers are not being added.

6. **Notifications Settings**: Review the notification settings in the repository. Make sure that notifications for pull requests and reviews are enabled for your team members. They can configure their notification settings in their GitHub account settings as well.

7. **GitHub Enterprise Compatibility**: If you're using GitHub Enterprise, ensure that the `uhg-actions/add-reviewers` action is compatible with your version of GitHub Enterprise and any specific configurations you have.

8. **Action Updates**: Check if there have been any updates or changes to the `uhg-actions/add-reviewers` action. It's possible that there have been updates that could affect its behavior.

9. **GitHub Support**: If the issue persists, consider reaching out to GitHub Support for assistance. They can help investigate and provide guidance on resolving the problem.

Remember that GitHub Actions and third-party actions can sometimes have interactions or dependencies that may impact their behavior. Troubleshooting such issues might require a combination of checking configuration settings, workflow logs, and action documentation.

**Reply by queenofcorgis** [reply]
@salini8642 hi there 👋🏼 , we want to encourage members to mark responses that helped solve their problem so please refrain from asking for a post to be marked as an answer - we want to make sure all our answers are providing the best solution possible. And I had a quick question:  just out of curiosity did you use ChatGPT or another similar LLM to write this answer? We're still working through how we want to include ChatGPT in the community so asking our users helps us learn more!
